addons/sale/models/crm_team.py

```diff
@@ -133,3 +145,13 @@ def action_primary_channel_button(self):
 
     def update_invoiced_target(self, value):
         return self.write({'invoiced_target': round(float(value or 0))})
+
+    @api.ondelete(at_uninstall=False)
+    def _unlink_except_if_used_for_sales(self):
```

Comment:
Other methods do not seem to add the "if".
I think "_unlink_except_used_for_sales" is sufficient. 

Side note: it's not explained here nor in the commit message why we prevent the deletion only for sales order, and also why "5" ?
I'm guessing because it's a core model and we want to protect it against undesired deletions that would be non-revertable blablabla ?

PS: "4 was too few and 6 is too many" will not be considered as an acceptable explanation :D

Reply:
It is a random number decided with Lna and Tfr. More than "just playing with odoo to discover it" and less than "really using it for my sales" :) . Will update commit message accordingly.

Currently indeed only sales orders are considered, because we consider those as "core" documents. Will update explanations and method naming.